PosteAPI/views.py

```diff
@@ -83,6 +81,29 @@ def post(self, request, *args, **kwargs):
                 {"result": {"success": False, "errors": serializer.errors}},
                 status=status.HTTP_400_BAD_REQUEST,
             )
+          
+    def update(self, request, *args, **kwargs):
+        user = request.user
+        data = request.data
+        folder = Folder.objects.get(['folder_id'])
+        if not user.can_share_folder(folder):
+            return Response({"detail": "You do not have permission to share this folder."},
+                            status=status.HTTP_403_FORBIDDEN)
+
+        permission, created = FolderPermission.objects.get_or_create(
+            user=User.objects.get(data['user_id']),
+            folder=folder,
+            # @TODO verify if I need to do any type casting
+            permission=data['permission']
+        )
+
+        # If the permission already exists, update it
+        if not created:
+            permission.permission = data['permission']
+            permission.save()
+
+        return Response({"detail": "Permission upsert successfully."},
```

Comment:
Typo?

Reply:
Nope - upsert is a portmanteau; it's a database operation that will update an existing row if a specified value already exists in a table, and insert a new row if the specified value doesn't already exist.